### PR TITLE
Pen 1378 video promo block

### PR DESCRIPTION
**Pull request by amyerz**
## Description
- update customfields for video promo blocks
- add inherit global content check box
- update README

## Jira Ticket
- [PEN-1378](https://arcpublishing.atlassian.net/browse/PEN-1378)

## Acceptance Criteria
- [ ] The group name Configure Video should be renamed to Configure Content (because we want to use this name consistently across all blocks so users always know where to configure content)

- [ ] An additional checkbox Custom Field should be added to the Configure Content group called Inherit global content
When this is checked true, the block should inherit the page’s global content

- [ ] The field Video UUID should be removed from the block completely (because this is duplicative – the video’s ID can be entered in the content source ID field)

- [ ] The Ratio custom field should be removed from the block completely (there’s no need for users to change video ratio) 

- [ ] The Autoplay custom field should be moved to a new group called Video settings

- [ ] The group named Content should be renamed Display settings

## Test Steps
- on the theme repo, update `.env` file:
  - link your local engine-theme-sdk, e.g. `ENGINE_SDK_REPO={PATH/TO/LOCAL/engine-theme-sdk}`
  - add `videoOrg=prod`
  - add `videoOrg=corecomponents` 
  - go to environment/index.json file and add `"videoEnv": "prod",`

- under block repo
  - create a new page and add video promo block
  
--- 
Test against content configurations:
- [ ] make sure inherit global content checkbox is UNCHECKED 
- [ ] update the `website_url` with `/video/2020/09/21/a-shallow-river-streaming-through-a-bed-of-rocks/` under customfields
- [ ] click the update button
- [ ] confirm the video match with https://corecomponents.arcpublishing.com/videocenter/video/v2/#/clip/5dc1d1c252faff00098c3399

--- 
Test against global content
- [ ] update global content on the page with `website_url` with `/video/2019/12/17/a-snowy-waterfall/`
Screen Shot 2020-11-19 at 3 48 10 PM
- [ ] confirm the videomatch with https://corecomponents.arcpublishing.com/videocenter/video/v2/#/clip/5df2b5164cedfd00092d093c

---
Test Inherit Global content
- [ ] once the content config and global content is set up as above
- [ ] click the Inherit Global Content button and it should toggle between above videos

--- 
Test Display Settings
- [ ] add "title" under "Display setting
- [ ] add "description" under "display setting"
- [ ] both title and description will display on screen

### Before
- customfields include ratio, video uuid
Screen Shot 2020-11-05 at 2 54 58 PM

### After
Screen Shot 2020-11-19 at 5 36 34 PM

 

[include screenshot or gif or link to video, storybook would be awesome]

## Dependencies or Side Effects
_Examples of dependencies or side effects are:_
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Update to css framework or SDK
- Dependency on another PR that needs to be merged first

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [ ] Confirmed all the test steps above are working
- [ ] Confirmed there are no linter errors
- [ ] Confirmed this PR has reasonable code coverage
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

### CODE REVIEW

**Review by JackHowa — CHANGES_REQUESTED** [review]
@amyerz can you handle the merge conflicts?

**Comment by amyerz** [comment]
- resolved conflicts
- updated customfields UI added playthrough:
Screen Shot 2020-11-13 at 4 43 06 PM

**Comment by JackHowa** [comment]
still don't see any video. have tried running npm install in the video promo block

Screen Shot 2020-11-16 at 15 19 48
Screen Shot 2020-11-16 at 15 19 34
Screen Shot 2020-11-16 at 15 16 58

**Review by JackHowa — CHANGES_REQUESTED** [review]
Still don't see any videos -- not sure if I'm missing something. I have put my steps that I used

**Comment by JackHowa** [comment]
Screen Shot 2020-11-16 at 15 22 04

**Review by JackHowa — CHANGES_REQUESTED** [review]
Turns out, there were fusion-news-theme environment dependencies for this pr. Amy is updating instructions here and adding a pr against fusion-news-theme as well

**Comment by amyerz** [comment]
Update instructions:
1. under the fusion-news-theme repo, add the followings to the `.env` file:
`videoOrg=prod`
`videoOrg=corecomponents`
and link to your local engine-sdk-theme repo
`ENGINE_SDK_REPO={local-path}/blocks/engine-theme-sdk`

2. under  fusion-news-theme repo, update `environment/index.json`
update/ add `  "videoEnv": "prod",`

**Comment by amyerz** [comment]
updated README under different PR: https://github.com/WPMedia/fusion-news-theme-blocks/pull/553/commits/f4ed17e8f0be7d195df72e83b75e8abce2dc9a6c

**Review by JackHowa — APPROVED** [review]

**Comment by JackHowa** [comment]
@amyerz can you resolve these conflicts now that I've merged the other pr

**Review by JackHowa — CHANGES_REQUESTED** [review]
fix merge conflicts with the other video block pr

**Comment by amyerz** [comment]
updated & resolved conflicts

File: blocks/video-promo-block/features/video-promo/default.jsx [hunk 1]
```diff
@@ -28,26 +30,40 @@ const AlertBadge = styled.span`
...
-  const content = uuid ? null : useContent({
+  // can be overwrite by globalContent
+  let {
+    title,
+    description,
+  } = customFields;
+
+  const { globalContent = {} } = useFusionContext();
+
+  const content = inheritGlobalContent ? globalContent : useContent({
```
Comment (JackHowa) [inline blocks/video-promo-block/features/video-promo/default.jsx]: can you try to fix this conditional `useContent`? we talked about trying not to add more react-hooks warning in the meeting. see here for more info https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

**Review by JackHowa — CHANGES_REQUESTED** [review]
We talked about in the linting meeting not adding any more rule of hooks violations. I added a comment in there with documentation to help

File: blocks/video-promo-block/features/video-promo/default.jsx (hunk 1 above)
Comment (amyerz) [reply blocks/video-promo-block/features/video-promo/default.jsx]: updated & test coverage updates
Screen Shot 2020-11-19 at 12 10 51 AM

**Comment by scarothers** [comment]
I noticed that one of your testing steps @amyerz is to add **/news/2020/10/20/this-story-has-a-video-as-promo-itemsbasic/** as global content – I actually don't think this is expected to work. That object is a **type: story**, not video. (It has a video in **promo_items.lead_art**, but rendering that is the responsibility of the Lead Art Block, not this block). 

To test with Global Content, a better URL to use is: **/video/2019/12/12/castle-church-tower/** – this is actually a **type: video** object. It's what we use as the example Global Content on the video template in Core Components https://corecomponents.arcpublishing.com/pagebuilder/editor/curate?t=video&v=v089DowxOVktKr

Let me know if that makes sense or if you have any questions about it – cc @JackHowa

**Comment by JackHowa** [comment]
seeing not found on the example test
Screen Shot 2020-11-19 at 11 11 26
 page https://corecomponents.arcpublishing.com/pagebuilder/editor/curate?t=video&v=v089DowxOVktKr

**Comment by JackHowa** [comment]
Not able to validate with the website_url /video/2019/12/12/castle-church-tower/  ... seeing 404. can you look into this @amyerz ? 

Screen Shot 2020-11-19 at 11 15 34

also the snowy waterful example story is showing, which I believe is unexepcted behavior 
Screen Shot 2020-11-19 at 11 14 34

**Review by JackHowa — CHANGES_REQUESTED** [review]
@amyerz Not able to validate based on new test instructions from @scarothers

**Comment by scarothers** [comment]
@JackHowa I'm sorry, that's my fault, I should have specified that this URL needs to be tested with The Gazette website. (The video is not published to The Sun or Arc Demo 5, which is why you see errors there.) Can you re-try with The Gazette selected in the admin? 

Screen Shot 2020-11-19 at 1 46 46 PM

**Comment by JackHowa** [comment]
>`I noticed that one of your testing steps @amyerz is to add /news/2020/10/20/this-story-has-a-video-as-promo-itemsbasic/ as global content – I actually don't think this is expected to work.` 

should this work? like, is it part of the ac for this not to render?

**Comment by scarothers** [comment]
When I add **/news/2020/10/20/this-story-has-a-video-as-promo-itemsbasic/** as the Global Content on the Video template now, it looks like this: 
Screen Shot 2020-11-19 at 2 34 23 PM
(again that's using The Gazette website) 

Basically, the video does not render. This is expected behavior, because that URL is for a **type: story** object, and this block is meant to render **type: video** objects. I would expect it to look the same once this change is complete.

File: blocks/video-promo-block/features/video-promo/default.jsx [hunk 2]
```diff
@@ -18,36 +20,58 @@ const DescriptionText = styled.p`
...
 const VideoPromo = ({ customFields }) => {
   const { arcSite } = useFusionContext();
   const {
-    uuid,
     autoplay = false,
-    ratio = 0.5625,
-    title,
-    description,
+    inheritGlobalContent = true,
     playthrough = false,
```
Comment (JackHowa) [inline blocks/video-promo-block/features/video-promo/default.jsx]: playthrough looks like it's default is already false based on default proptypes. you can remove the = false
```suggestion
    playthrough,
```
Screen Shot 2020-11-19 at 14 22 41

File: blocks/video-promo-block/features/video-promo/default.jsx [hunk 3]
```diff
@@ -18,36 +20,58 @@ const DescriptionText = styled.p`
...
 
 const VideoPromo = ({ customFields }) => {
   const { arcSite } = useFusionContext();
   const {
-    uuid,
     autoplay = false,
-    ratio = 0.5625,
-    title,
-    description,
+    inheritGlobalContent = true,
```
Comment (JackHowa) [inline blocks/video-promo-block/features/video-promo/default.jsx]: based on the proptypes, it looks like we already have `true` 

```suggestion
    inheritGlobalContent,
```
Screen Shot 2020-11-19 at 14 23 08

File: blocks/video-promo-block/features/video-promo/default.jsx [hunk 4]
```diff
@@ -18,36 +20,58 @@ const DescriptionText = styled.p`
...
-  color: #FFFFFF;
-  background-color: #DB0A07;
-  border-radius: 5px;
 `;
 
 const VideoPromo = ({ customFields }) => {
   const { arcSite } = useFusionContext();
   const {
-    uuid,
     autoplay = false,
```
Comment (JackHowa) [inline blocks/video-promo-block/features/video-promo/default.jsx]: autoplay is already default false based on proptypes 

Screen Shot 2020-11-19 at 14 24 34
```suggestion
    autoplay,
```

**Comment by amyerz** [comment]
Note: 
update the page's global content with `/video/2019/12/17/a-snowy-waterfall/`
Screen Shot 2020-11-19 at 3 48 10 PM

and update config's content in customfield description: `/video/2020/09/21/a-shallow-river-streaming-through-a-bed-of-rocks/`
Screen Shot 2020-11-19 at 3 47 57 PM

**Comment by amyerz** [comment]
updated:
Screen Shot 2020-11-19 at 5 36 34 PM

**Review by JackHowa — APPROVED** [review]
good work on this

**Comment by JackHowa** [comment]
able to validate toggling between the two content sources in the test instructions 

Screen Shot 2020-11-20 at 13 10 51


Screen Shot 2020-11-20 at 13 10 58

**Pull request merged**